docs/scripting-ref.rst

```diff
@@ -391,7 +393,225 @@ Compare:
         -- ...
     end
 
-See also: :ref:`splash-runjs`, :ref:`splash-jsfunc`, :ref:`splash-autoload`.
+See also: :ref:`splash-runjs`, :ref:`splash-jsfunc`, :ref:`splash-autoload`,
+:ref:`wait-for-resume`.
+
+.. _wait-for-resume:
+
+splash:wait_for_resume
+----------------------
+
+Run asynchronous JavaScript code in page context. The Lua script will
+yield until the JavaScript code tells it to resume.
+
+**Signature:** ``result, error = splash:wait_for_resume(snippet, timeout)``
+
+**Parameters:**
+
+* snippet - a string with a JavaScript source code to execute. This code
+  must include a function called ``main``. The first argument to ``main``
+  is an object that has the properties ``resume`` and ``error``. ``resume``
+  is a function which can be used to resume Lua execution. It takes an optional
+  argument which will be returned to Lua in the ``result`` return value.
+  ``error`` is a function which can be called with a required string value
+  that is returned in the ``error`` return value.
+* timeout - a number which determines (in seconds) how long to allow JavaScript
+  to execute before forceably returning control to Lua. Defaults to
+  zero, which disables the timeout.
+
+**Returns:** ``result, error`` pair. When the execution is successful
+``result`` is a table. If the value returned by JavaScript is not
+``undefined``, then the ``result`` table will contain a key ``value``
+that has the value passed to ``splash.resume(…)``. The ``result`` table also
+contains any additional key/value pairs set by ``splash.set(…)``. In case of
+timeout or JavaScript errors ``result`` is ``nil`` and ``error`` contains an
+error message string.
+
+Examples:
+
+The first, trivial example shows how to transfer control of execution from Lua
+to JavaScript and then back to Lua. This command will tell JavaScript to
+sleep for 3 seconds and then return to Lua. Note that this is an async
+operation: the Lua event loop and the JavaScript event loop continue to run
+during this 3 second pause, but Lua will not continue executing the current
+function until JavaScript calls ``splash.resume()``.
+
+.. code-block:: lua
+
+    function main(splash)
+
+        local result, error = splash:wait_for_resume([[
+            function main(splash) {
+                setTimeout(function () {
+                    splash.resume();
+                }, 3000);
+            }
+        ]])
+
+        -- result is {}
+        -- error is nil
+
+    end
+
+``result`` is set to an empty table to indicate that nothing was returned
+from ``splash.resume``. You can use ``assert(splash:wait_for_resume(…))``
+even when JavaScript does not return a value because the empty table signifies
+success to ``assert()``.
+
+.. note::
+
+    The ``main()`` function is required, and you will get an error if
```

Comment:
There are two 'main' functions in the example above. The note applies to both, but it could make sense to stress that the note is about JS `main` function.